### ISSUE TITLE
Installation with setup.py fails

### ISSUE THREAD

**Issue by visola** [rycus86/prometheus_flask_exporter#90]
I have a Python app and I'm trying to add the exporter but it fails to install using setuptools. The code lives in this repo: https://github.com/VinnieApps/my_finances

I added this to my `setup.py`: `'prometheus-flask-exporter==0.18.1',` and when I run `python setup.py install` I get the following error:

```
Running prometheus_flask_exporter-0.18.1/setup.py -q bdist_egg --dist-dir /var/folders/p9/8v8v0r4x4k91mx_bgjh4_5xm0000gn/T/easy_install-h1j4ojgx/prometheus_flask_exporter-0.18.1/egg-dist-tmp-cbtps1wd
error: Setup script exited with error: SandboxViolation: mkdir('/Users/visola/Library/Caches/com.apple.python/private/var/folders/p9/8v8v0r4x4k91mx_bgjh4_5xm0000gn/T/easy_install-h1j4ojgx', 511) {}

The package setup script has attempted to modify files on your system
that are not within the EasyInstall build area, and has been aborted.

This package cannot be safely installed by EasyInstall, and may not
support alternate installation locations even if you run its setup
script by hand.  Please inform the package's author and the EasyInstall
maintainers to find out if a fix or workaround is available.
```

I'm not an experience Python developer so any help would be greatly appreciated.

I normally execute all python commands in a virtual environment created with `venv`.

A bit more info as I debug this:

I'm on Mac OS X: 10.15.7

From within my venv:
```
$ which python
/Users/visola/git/my_finances/venv/bin/python

$ python --version
Python 3.8.2
```

**Comment by visola** [comment]
After reverting my changes and trying to recreate my virtual environment from scratch, it sounds like the problem is actually some Python conflict in my Mac OS. I'll close this issue for now.
If I find the solution, I'll post it here later.
Sorry for the noise.

**Comment by rycus86** [comment]
No worries, glad you managed to resolve your problem!

**Comment by visola** [comment]
For posterity, it seems like this is an issue with Mac OS. I fixed this by installing the latest Python 3 from Homebrew:

```
$ brew reinstall python3
```
1. Then restarting my terminal to ensure that I'm running the new python3. 
2. Removed all cached files and virtual environment related files. 
3. Recreated everything from scratch with `python3 -m venv venv` and 
4. reinstalled my packages:

```
$ source venv/bin/activate
$ python setup.py install
```

Now it works.
